packages/ui-patterns/form/Layout/FormLayout.tsx

```diff
@@ -0,0 +1,401 @@
+import { cva } from 'class-variance-authority'
+import React from 'react'
+import {
+  FormDescription_Shadcn_,
+  FormLabel_Shadcn_,
+  FormMessage_Shadcn_,
+  Label_Shadcn_,
+  cn,
+} from 'ui'
+import { SIZE } from 'ui/src/lib/constants'
+
+type Props = {
+  align?: 'left' | 'right'
+  description?: string | React.ReactNode
+  error?: string | React.ReactNode
+  label?: string | React.ReactNode
+  labelOptional?: string | React.ReactNode
+  layout?: 'horizontal' | 'vertical' | 'flex'
+  size?: 'tiny' | 'small' | 'medium' | 'large' | 'xlarge'
+  beforeLabel?: string | React.ReactNode
+  afterLabel?: string | React.ReactNode
+  nonBoxInput?: boolean
+  labelLayout?: 'horizontal' | 'vertical'
+  isReactForm?: boolean
+  hideMessage?: boolean
+  name?: string
+}
+
+const ContainerVariants = cva('grid gap-2', {
+  variants: {
+    size: {
+      tiny: 'text-xs',
+      small: 'text-sm leading-4',
+      medium: 'text-sm',
+      large: 'text-base',
+      xlarge: 'text-base',
+    },
+    align: { left: '', right: '' },
+    responsive: {
+      true: '',
+      false: '',
+    },
+    layout: {
+      horizontal: 'flex flex-row gap-6 justify-between',
+      vertical: 'flex flex-col gap-3',
+      flex: 'flex flex-row gap-3',
+    },
+    flex: {
+      true: '',
+      false: '',
+    },
+  },
+  compoundVariants: [
+    {
+      layout: 'flex',
+      align: 'right',
+      className: 'justify-between',
+    },
+  ],
+  defaultVariants: {},
+})
+
+const LabelContainerVariants = cva('', {
+  variants: {
+    flex: {
+      true: '',
+      false: '',
+    },
+    align: {
+      left: '',
+      right: '',
+    },
+    layout: {
+      horizontal: 'flex flex-col gap-2',
+      vertical: 'flex flex-row gap-2 justify-between',
+      flex: 'flex flex-col gap-0',
+    },
+    labelLayout: {
+      horizontal: '',
+      vertical: '',
+      ['']: '',
+    },
+  },
+  compoundVariants: [
+    {
+      flex: true,
+      align: 'left',
+      className: 'order-2',
+    },
+    {
+      flex: true,
+      align: 'right',
+      className: 'order-1',
+    },
+    {
+      layout: 'vertical' || 'flex',
```

Comment:
@MildTomato What's going on with this line and line 138?

Reply:
looks like line 96 is a layout variant for LabelContainer, and line 138 is a layout variant for the Container.

Reply:
But `"vertical" || "flex"` is always just `"vertical"` - what is `|| "flex"` supposed to do?